test/test_oneagentmetadataenricher.py

```diff
@@ -58,10 +75,122 @@ def test_mock_get_metadata_file(self, mock_func):
+        self.assertEqual("v1", res["k1"])
+        self.assertEqual("v2", res["k2"])
+
+    @patch('dynatrace.opentelemetry.metrics.export.oneagentmetadataenricher'
+           '.OneAgentMetadataEnricher._get_metadata_file_content')
+    def test_parse_empty_list(self, mock_func):
+        mock_func.return_value = []
+
+        enricher = OneAgentMetadataEnricher()
+        res = enricher._parse_oneagent_metadata(
+            enricher._get_metadata_file_content())
+
+        self.assertEqual(0, len(res))
+
+
+class TestGetIndirectionFile(unittest.TestCase):
+    def test_get_indirection_file_contents_success(self):
+        # when open is called, a file with this content is returned
+        mock = mock_open(read_data="metadata_file_name")
+        enricher = OneAgentMetadataEnricher()
+        with patch("builtins.open", mock):
+            res = enricher._get_metadata_file_name("not-used-since-mocking")
+            self.assertEqual("metadata_file_name", res)
+
+    def test_get_indirection_file_missing_or_not_readable(self):
+        # will throw an IOError when calling open
+        mock = mock_open()
+        mock.side_effect = IOError()
+        enricher = OneAgentMetadataEnricher()
+        with patch("builtins.open", mock):
+            res = enricher._get_metadata_file_name("not-used-since-mocking")
+            self.assertEqual(None, res)
+
+    def test_get_indirection_fname_missing_empty_or_invalid(self):
+        enricher = OneAgentMetadataEnricher()
+        self.assertEqual(None, enricher._get_metadata_file_name(""))
+        self.assertEqual(None, enricher._get_metadata_file_name(None))
+        self.assertEqual(None, enricher._get_metadata_file_name("#%&^:"))
+
+
+class TestGetMetadataContents(unittest.TestCase):
+    def test_indirection_file_empty(self):
+        mock = mock_open(read_data="")
+        enricher = OneAgentMetadataEnricher()
+
+        with patch("builtins.open", mock):
+            res = enricher._get_metadata_file_content()
+            self.assertEqual(0, len(res))
+
+    def test_get_file_contents_success(self):
+        mock = mock_open(read_data="key1=value1\nkey2=value2")
+        enricher = OneAgentMetadataEnricher()
+        with patch("builtins.open", mock):
+            res = enricher._get_metadata_file_content()
+            self.assertEqual(2, len(res))
+
+    @patch('dynatrace.opentelemetry.metrics.export.oneagentmetadataenricher'
+           '.OneAgentMetadataEnricher._get_metadata_file_name')
+    def test_get_file_contents_from_none(self, mock_func):
+        mock_func.return_value = None
+
+        enricher = OneAgentMetadataEnricher()
+        res = enricher._get_metadata_file_content()
+        self.assertEqual(0, len(res))
+
+    @patch('dynatrace.opentelemetry.metrics.export.oneagentmetadataenricher'
+           '.OneAgentMetadataEnricher._get_metadata_file_name')
+    def test_get_file_contents_from_tmpfile(self, mock_func):
+        with tempfile.TemporaryDirectory() as tmp_dir:
```

Comment:
Does this clean up the directory afterwards or will you have to delete the file manually?

Reply:
"On completion of the context or destruction of the temporary directory object the newly created temporary directory and all its contents are removed from the filesystem.", so it should be gone.